editor/editor_file_system.cpp

```diff
@@ -2112,61 +2139,66 @@ void EditorFileSystem::reimport_files(const Vector<String> &p_files) {
 
 	int from = 0;
 	for (int i = 0; i < reimport_files.size(); i++) {
-		if (use_multiple_threads && reimport_files[i].threaded) {
-			if (i + 1 == reimport_files.size() || reimport_files[i + 1].importer != reimport_files[from].importer) {
-				if (from - i == 0) {
-					//single file, do not use threads
-					pr.step(reimport_files[i].path.get_file(), i);
-					_reimport_file(reimport_files[i].path);
-				} else {
-					Ref<ResourceImporter> importer = ResourceFormatImporter::get_singleton()->get_importer_by_name(reimport_files[from].importer);
-					ERR_CONTINUE(!importer.is_valid());
-
-					importer->import_threaded_begin();
-
-					ImportThreadData tdata;
-					tdata.max_index = from;
-					tdata.reimport_from = from;
-					tdata.reimport_files = reimport_files.ptr();
-
-					WorkerThreadPool::GroupID group_task = WorkerThreadPool::get_singleton()->add_template_group_task(this, &EditorFileSystem::_reimport_thread, &tdata, i - from + 1, -1, false, vformat(TTR("Import resources of type: %s"), reimport_files[from].importer));
-					int current_index = from - 1;
-					do {
-						if (current_index < tdata.max_index) {
-							current_index = tdata.max_index;
-							pr.step(reimport_files[current_index].path.get_file(), current_index);
-						}
-						OS::get_singleton()->delay_usec(1);
-					} while (!WorkerThreadPool::get_singleton()->is_group_task_completed(group_task));
-
-					WorkerThreadPool::get_singleton()->wait_for_group_task_completion(group_task);
+		if (!groups_to_reimport.has(reimport_files[i].path)) {
```

Comment:
Avoid indenting everything like that, instead a negated condition and early `continue`/`break`/`return` is preferred (just like in #60888).